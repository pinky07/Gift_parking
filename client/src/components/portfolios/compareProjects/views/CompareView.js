import React from 'react';
import CompareViewSuccess from "./viewComponents/CompareViewSuccess";
import CompareWithCriticalError from "./viewComponents/CompareWithCriticalError";
import LoadingCompareView from "./viewComponents/LoadingCompareView";

const CompareView = ({viewModel}) => {
  const {errorMessage} = viewModel;

  if (errorMessage)
    return (<CompareWithCriticalError errorMessage={errorMessage}/>);
  else {
    const {portfolioName} = viewModel.comparison;
    if (portfolioName) {
      return (<CompareViewSuccess viewModel={viewModel}/>);
    }
    else {
      return (<LoadingCompareView/>);
    }
  }
}

export default CompareView;

