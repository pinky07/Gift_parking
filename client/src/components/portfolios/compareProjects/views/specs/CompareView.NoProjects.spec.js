import React from 'react';
import renderer from 'react-test-renderer';

import CompareView from '../CompareView';

describe('CompareView when portfolio does not have projects', () => {

  let viewModel;
  let actual;

  it('should render compare view with no projects', () => {

    viewModel = {
      portfolioId: '5',
      comparison: getPortfolioWithNoProjects(),
      errorMessage: '',
    };
    actual = renderer.create(<CompareView viewModel={viewModel}/>).toJSON();

    expect(actual).toMatchSnapshot();
  });

  function getPortfolioWithNoProjects() {
    // Test data from http://http://localhost:8080/api/v1/portfolios/5/comparison
    return {"portfolioName": "Portfolio With No Projects", "lastSnaps": []};
  }

});