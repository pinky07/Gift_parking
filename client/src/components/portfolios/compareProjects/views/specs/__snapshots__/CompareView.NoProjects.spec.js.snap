// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareView when portfolio does not have projects should render compare view with no projects 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <h1
    className="grommetux-heading"
  >
    Compare Projects for Portfolio: 
    Portfolio With No Projects
  </h1>
  <h3>
    <svg
      aria-label="Unknown"
      className="grommetux-status-icon grommetux-status-icon-unknown"
      role="img"
      version="1.1"
      viewBox="0 0 24 24"
    >
      <defs>
        <mask
          id="mask-uknown-0"
        >
          <g
            className="grommetux-status-icon__detail"
          >
            <rect
              fill="#fff"
              height="24"
              width="24"
              x="0"
              y="0"
            />
            <circle
              cx="12"
              cy="12"
              fill="#000"
              r="10"
              stroke="none"
            />
            <path
              d="M9,10.4 C9,8.8 10.4,7.6 12,7.6 C13.6,7.6 14.9,9 15,10.4 C15,11.7 14.1,12.7 12.9,13.1 C12.4,13.2 12,13.7 12,14.2 L12,15.5"
              fill="none"
              stroke="#fff"
              strokeWidth="2"
            />
            <circle
              cx="12"
              cy="17.6"
              fill="#fff"
              r="1"
              stroke="none"
            />
          </g>
        </mask>
      </defs>
      <g
        className="grommetux-status-icon__base"
        mask="url(#mask-uknown-0)"
      >
        <circle
          className="grommetux-status-icon__normal"
          cx="12"
          cy="12"
          r="12"
          stroke="none"
        />
        <circle
          className="grommetux-status-icon__small"
          cx="12"
          cy="12"
          fill="none"
          r="11"
          strokeWidth="3"
        />
      </g>
    </svg>
    <span>
      This Portfolio has no projects.
    </span>
  </h3>
</div>
`;
